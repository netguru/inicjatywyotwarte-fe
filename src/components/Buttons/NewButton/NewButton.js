import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

const AddButton = styled(Link)`
  color: ${props => props.theme.fontColorDark};
  font-size: 0.85rem;
  padding: 0 1rem;
  background-color: ${props => props.theme.addInitiativeButtonBackgroundColor};
  border: 1px solid ${props => props.theme.addInitiativeButtonBorderColor};
  border-radius: ${props => props.theme.borderRadiusBig};
  cursor: pointer;
  height: 2.5rem;
  background-color: ${props => props.theme.lightBackgroundColor};
  transition: ${props => props.theme.buttonTransition};
  text-decoration: none;
  display: flex;
  align-items: center;

  &:hover {
    background-color: ${props => props.theme.fontColorDark};
    color: ${props => props.theme.lightBackgroundColor};
  }
`
AddButton.displayName = 'AddButton';

export default function NewButton () {
  return <AddButton to='/dodaj-inicjatywe'>+ Dodaj inicjatywÄ™</AddButton>
}
