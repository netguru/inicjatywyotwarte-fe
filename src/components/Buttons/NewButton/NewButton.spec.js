import React from 'react'
import NewButton from './NewButton'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('NewButton', () => {
  test('renders button', () => {
    const wrapper = shallow(<NewButton/>);
    expect(wrapper.find('AddButton[to="/dodaj-inicjatywe"]').length).toEqual(1);
    expect(wrapper.find('AddButton[to="/dodaj-inicjatywe"]').text()).toEqual('+ Dodaj inicjatywÄ™');
  });

  test('match snapshot', () => {
    const wrapper = shallow(<NewButton/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


