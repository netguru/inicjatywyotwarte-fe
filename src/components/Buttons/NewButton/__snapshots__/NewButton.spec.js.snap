// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewButton match snapshot 1`] = `
<AddButton
  to="/dodaj-inicjatywe"
>
  + Dodaj inicjatywÄ™
</AddButton>
`;
